src/algebra/module/torsion.lean

```diff
@@ -433,20 +433,34 @@ section torsion
 variables [comm_semiring R] [add_comm_monoid M] [module R M]
 open_locale big_operators
 
-lemma is_torsion_by_ideal_of_finite_of_is_torsion [module.finite R M] (hM : module.is_torsion R M) :
-  ∃ I : ideal R, (I : set R) ∩ R⁰ ≠ ∅ ∧ module.is_torsion_by_set R M I :=
+variables (R M)
+/-- An ideal satisfying `module.is_torsion_by_set R M I`. -/
+noncomputable def _root_.module.torsion_ideal [module.finite R M] : ideal R :=
+∏ x in ‹module.finite R M›.out.some, ideal.torsion_of R M x
```

Comment:
This doesn't seem to be a particularly well-established name for this construction, but I can't find one elsewhere.

Is this construction maximal, in that it is the largest such ideal?

Reply:
Actually, the largest such ideal could also be defined as `annihilator ⊤` (without a `module.finite` hypothesis). Maybe it's a better idea to prove in a lemma `annihilator ⊤ = ⨅ x ∈ ‹module.finite R M›.out.some, ideal.torsion_of R M x` (true for any finite module) and then use it to prove `(annihilator ⊤ : set R) ∩ R⁰ ≠ ∅` for finite torsion modules.

Reply:
I actually didn't need `⨅ x ∈ ‹module.finite R M›.out.some, ideal.torsion_of R M x` at all here.